src/app/regular-giving/regular-giving.component.ts

```diff
@@ -62,33 +69,87 @@ export class RegularGivingComponent implements OnInit {
 
     this.campaign = this.route.snapshot.data.campaign;
     this.bannerUri$ = this.imageService.getImageUri(this.campaign.bannerUri, 830);
-    this.mandateForm = this.formBuilder.group([]);
+    this.mandateForm = this.formBuilder.group({
+        donationAmount: ['', [
+          requiredNotBlankValidator,
+          getCurrencyMinValidator(1), // for now min & max are hard-coded, will change to be based on a field on
+                                      // the campaign.
+          getCurrencyMaxValidator(500),
+          Validators.pattern('^\\s*[£$]?[0-9]+?(\\.00)?\\s*$'),
+        ]],
+      }
+    );
   }
 
   interceptSubmitAndProceedInstead(event: Event) {
     event.preventDefault();
     this.next();
   }
 
+  stepChanged(_event: StepperSelectionEvent) {
+    // no-op for now.
+  }
+
   next() {
     this.stepper.next();
   }
 
-  async stepChanged(_event: StepperSelectionEvent) {
-  }
+  submit() {
+    const invalid = this.mandateForm.invalid;
+    if (invalid) {
+      let errorMessage = 'Form error: ';
+      if (this.mandateForm.get('donationAmount')?.hasError('required')) {
+        errorMessage += "Monthly donation amount is required";
+      }
+      this.showError(errorMessage);
+      return;
+    }
 
-  logout() {
-    this.identityService.logout();
-    this.router.navigate(['']);
+    const donationAmountPounds = +this.mandateForm.value.donationAmount;
+    const amountInPence = donationAmountPounds * 100;
+
+    /**
+     * @todo consider if we need to send this from FE - if we're not displaying it to donor better for matchbot to
+     *       generate it.*/
+    const dayOfMonth = Math.min(new Date().getDate(), 28);
+
+    console.log(invalid)
+    if (invalid) {
+      this.showError("Form is not filled in correctly yet");
+      this.showError(JSON.stringify(this.mandateForm.errors))
+      return;
+    }
+
+
+    this.regularGivingService.startMandate({
+      amountInPence,
+      dayOfMonth,
+      campaignId: this.campaign.id,
+      currency: "GBP",
+      giftAid: false
+    }).subscribe({
+    next: (mandate: Mandate) => {
+      const prefix = "Regular giving mandate created: (todo - make 'thanks' page with polling for updates and/or make " +
+        "the mandate come back activated so it will show on my regular giving page) ";
+
+      alert(prefix + JSON.stringify(mandate));
```

Comment:
of course we won't keep the `alert` for long but I think it's OK to merge it as-is for now. It gives a relatively easy way to test that the mandate is created appropriately before we've either updated the status to active or built anything to display a pending mandate.